types/index.d.ts

```diff
@@ -0,0 +1,151 @@
+import * as stream from 'stream';
+import { EventEmitter } from 'events'
+
+
+export type Callback<V, Err = any> = (err: Err, value: V) => void;
+
+export type CodecTransformer = (buffer: Buffer, callback: () => void) => Buffer; // TODO
+
+export interface CodecOptions {
+	deflate: CodecTransformer;
```

Comment:
Is 8 spaces per tab a strong convention for typings? If not, can we switch to 2 for consistency with the other files?

Reply:
I have no strong feelings on file indentations, although I think that two spaces provide not enough visual contrast. I will  provide an extra commit.